math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -100,27 +120,945 @@ auto MapFromTuple(Tuple_t &&t, std::index_sequence<Is...>)
 
 namespace Internal {
 namespace VecOps {
+/// Return the next power of two (in 64-bits) that is strictly greater than A.
+/// Return zero on overflow.
+inline uint64_t NextPowerOf2(uint64_t A)
+{
+   A |= (A >> 1);
+   A |= (A >> 2);
+   A |= (A >> 4);
+   A |= (A >> 8);
+   A |= (A >> 16);
+   A |= (A >> 32);
+   return A + 1;
+}
+
+/// This is all the stuff common to all SmallVectors.
+///
+/// The template parameter specifies the type which should be used to hold the
+/// Size and Capacity of the SmallVector, so it can be adjusted.
+/// Using 32 bit size is desirable to shrink the size of the SmallVector.
+/// Using 64 bit size is desirable for cases like SmallVector<char>, where a
+/// 32 bit size would limit the vector to ~4GB. SmallVectors are used for
+/// buffering bitcode output - which can exceed 4GB.
+template <class Size_T>
+class SmallVectorBase {
+protected:
+   void *BeginX;
+   Size_T fSize = 0, Capacity;
```

Comment:
To clearly see memory errors, consider
```suggestion
   Size_T fSize = 0;
   Size_T Capacity = -1;
```
To only make the declaration easier to read, consider
```suggestion
   Size_T fSize = 0;
   Size_T Capacity;
```